blockchain/filedao/staging_buffer.go

```diff
@@ -15,41 +17,53 @@ import (
 
 type (
 	stagingBuffer struct {
+		lock   sync.RWMutex
 		size   uint64
+		start  uint64
 		buffer []*block.Store
-		deser  *block.Deserializer
 	}
 )
 
-func newStagingBuffer(size uint64, deser *block.Deserializer) *stagingBuffer {
+func newStagingBuffer(size, start uint64) *stagingBuffer {
 	return &stagingBuffer{
 		size:   size,
+		start:  start,
 		buffer: make([]*block.Store, size),
-		deser:  deser,
 	}
 }
 
-func (s *stagingBuffer) Get(pos uint64) (*block.Store, error) {
-	if pos >= s.size {
+func (s *stagingBuffer) Get(height uint64) (*block.Store, error) {
+	if height < s.start {
 		return nil, ErrNotSupported
 	}
-	return s.buffer[pos], nil
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+	return s.buffer[s.slot(height)], nil
 }
 
-func (s *stagingBuffer) Put(pos uint64, blk *block.Store) (bool, error) {
-	if pos >= s.size {
+func (s *stagingBuffer) Put(height uint64, blk *block.Store) (bool, error) {
+	if height < s.start {
 		return false, ErrNotSupported
 	}
+	pos := s.slot(height)
+	s.lock.Lock()
+	defer s.lock.Unlock()
 	s.buffer[pos] = blk
 	return pos == s.size-1, nil
 }
 
+func (s *stagingBuffer) slot(height uint64) uint64 {
+	return (height - s.start) % s.size
+}
+
 func (s *stagingBuffer) Serialize() ([]byte, error) {
 	blkStores := []*iotextypes.BlockStore{}
 	// blob sidecar data are stored separately
+	s.lock.Lock()
```

Comment:
seems unnecessary, b/c `Put` and `Serialize` will not be called parallelly

Reply:
1. in current code, they are not called parallelly. As 2 public methods, should consider synchronize access?
2. `Serialize` is actually also a read operation, so can change to `RLock`